lib/internal/modules/cjs/loader.js

```diff
@@ -606,9 +606,20 @@ Module._resolveFilename = function(request, parent, isMain, options) {
   // Look up the filename first, since that's the cache key.
   var filename = Module._findPath(request, paths, isMain);
   if (!filename) {
+    const requireStack = [];
+    for (var cursor = parent;
+      cursor;
+      cursor = cursor.parent) {
+      requireStack.push(cursor.filename || cursor.id);
+    }
+    let message = `Cannot find module '${request}'`;
+    if (requireStack.length > 0) {
+      message = message + '\nRequire stack:\n- ' + requireStack.join('\n- ');
+    }
     // eslint-disable-next-line no-restricted-syntax
-    var err = new Error(`Cannot find module '${request}'`);
+    var err = new Error(message);
     err.code = 'MODULE_NOT_FOUND';
+    err.requireStack = requireStack;
```

Comment:
Can you add documentation about this in `doc/api/errors.md`?

Reply:
Done.